docs/kns_apps_tutorial.md

```diff
@@ -0,0 +1,347 @@
+
+### 2.2. Ingress parameters
+
+Ingress parameters allow to connect app to the external world, by giving final user the URL to reach the app portal/entry point/API/etc.
+
+```json
+                "ingress_type": "subdomain",
+```
+
+`ingress_type` specify the type of ingress to be use. By default it is `subdomain`, which means that for each job, the ingress URL will be of format kns-job-XXX.mycluster.example.
+On some clusters, a subdomain based ingress cannot be used, and so instead of providing a subdomain based ingress, the url will be of format mycluster.example/XXX. To do so, set `ingress_type` to `prefix`. Note that not all KNS applications are compatible with this workaround (Kubeflow for example will not work).
+
+```json
+                "targets": [
+                    {
+                        "service_name": "web",
+                        "service_port": 8080,
+                        "service_namespace": "default"
+                    }
+                ],
+```
+
+`targets` list is then used to defind ingresses to spawn.
+For now, KNS will only consider the first item in the list. This limitation will be removed in later versions.
+
+The following parameters are to be set for each ingress target:
+
+* `service_name`: name of the service to link ingress to.
+* `service_port`: port of the service to link ingress to.
+* `service_namespace`: namespace where the service to link ingress to is.
+
+### 2.3. Templates parameters
+
+#### 2.3.1. Global settings
+
+```json
+                "templates_settings": {
+                    "global_retry": 20,
+                    "global_retry_sleep": 15
+                },
+```
+
+Templates are provided as a list, named `templates`, which will be executed in an ordered manner.
+
+While each item of the list can set its own parameters, the following global parameters are available, which apply to all items of the tempate list:
+
+* `global_retry`: how much retry to do if executing the item failed. For example, if the kubectl apply of the template fail, it will be retried. This is useful when a step needs another previous one to have completed to succeed (resources running).
+* `global_retry_sleep`: how much seconds should we wait between each retry.
+
+#### 2.3.2. Templates
+
+```json
+                "templates": [
+                    {
+                        "name": "google-sample",
+                        "type": "kubectl",
+                        "path": "google-sample.yml"
+                    }
+                ]
+```
+
+The `templates` list allow to execute instructions in an ordered manner, and to apply templates to deploy the app into the nested scheduler.
+Since the init pod runs with cluster-admin rights (inside the nested K3S cluster), templates can create new namespaces, cluster roles, etc.
```

Comment:
Fist reference to init pod. This has not been explained before. Maybe it would be better to talk in general about KNS without references to internal components.

Reply:
Good point, I removed this sentence.